src/backend/distributed/utils/aggregate_utils.c

```diff
@@ -400,13 +466,35 @@ worker_partial_agg_sfunc(PG_FUNCTION_ARGS)
 		}
 	}
 
-	InitFunctionCallInfoData(*innerFcinfo, &info, fcinfo->nargs - 1, fcinfo->fncollation,
-							 fcinfo->context, fcinfo->resultinfo);
-	fcSetArgExt(innerFcinfo, 0, box->value, box->valueNull);
-	for (argumentIndex = 1; argumentIndex < innerFcinfo->nargs; argumentIndex++)
+	if (combinedInputDesc != NULL)
```

Comment:
As far as I understand, we do this `if/else` for backward compatibility. Might be useful to note  which Citus versions this trick is necessary (so that we can remove this part of the code in the next few releases?)

Overall, is it feasible to split this function into two, where one function handles `RowExpr` and the other handles the old logic? So, we'd have something like:

```C

worker_partial_agg_sfunc():
     do_some_common_checks()
     do_some_common_start_up_logic()
      if (rowExpr)
          new_logic_function()
     else
         old_logic_function()
```

With that, it'd be much easier to follow the code and retire the old logic when needed.

(Similarly  for `TypecheckWorkerPartialAggArgType `)